compiler/rustc_hir_typeck/src/generator_interior/mod.rs

```diff
@@ -92,6 +94,13 @@ impl<'a, 'tcx> InteriorVisitor<'a, 'tcx> {
                 Some(YieldData { span: DUMMY_SP, expr_and_pat_count: 0, source: self.kind.into() })
             });
 
+        // If this is the `&mut Context<'_>` async resume argument, or we are
+        // just visiting a `_task_context = yield ()` expression from async
+        // lowering, we do *not* consider this type to be live across yields.
+        if Some(hir_id) == self.task_context_hid || self.in_await_yield {
```

Comment:
Is it possible to more strongly assert that the thing that we're intentionally *not* considering live across the yield is a `&mut Context<'_>`? Maybe assert on the value of `ty: Ty<'tcx>` above?

Reply:
How do I best assert that though without manually constructing a matching type? I could probably get the expected resume_ty from the generator somehow and use that?

Reply:
Can't you destructure the type, check if it's a mutable reference to a specific ADT? The Context type is a lang item, right?

Reply:
Yes, did that now, though its a bit more complicated than hoped.

Did also rename some things and added more comments.